common/schema-model/src/main/java/io/smallrye/graphql/schema/model/Scalars.java

```diff
@@ -181,6 +188,11 @@ public static void addUuid() {
         // Void
         populateScalar(Void.class.getName(), VOID, Void.class.getName());
         populateScalar(void.class.getName(), VOID, Void.class.getName());
+
+        // Federation
+        populateScalar(FieldSet.class.getName(), FIELD_SET, FieldSet.class.getName());
+        populateScalar(PolicyItem.class.getName(), POLICY, PolicyItem.class.getName());
+        populateScalar(ScopeItem.class.getName(), SCOPE, ScopeItem.class.getName());
```

Comment:
I had to remove the `Boolean.getBoolean("smallrye.graphql.federation.enabled")` check here because variable `io.smallrye.graphql.schema.model.Scalars#scalarMap` is static. When running tests inside `server/implementation/src/test/java/io/smallrye/graphql`, the `Scalars` class is initialized without federation classes, since federation is disabled by default. However, when running `io.smallrye.graphql.schema.SchemaTest#testSchemaWithFederationEnabled`, federation is enabled, but the `scalarMap` remains unchanged.

Reply:
If you remove it, then these directives will always be in the schema. I'm not sure I understand your explanation for removing it - why exactly? Is it that `io.smallrye.graphql.schema.SchemaTest#testSchemaWithFederationEnabled` doesn't see them even though it sets `smallrye.graphql.federation.enabled=true` at the beginning? I wonder why this was working well before and now it isn't?!

Reply:
Ah right, now I understand your point. This is a bit tricky because the static initializer will be run just once, unless we run the test with a separate JVM or class loader. Let me think...

Reply:
Could you split `io.smallrye.graphql.schema.SchemaTest` into two, where you introduce `io.smallrye.graphql.schema.SchemaTestFederationEnabled` that just contains the `testSchemaWithFederationEnabled` method (plus whatever it uses), and make sure that each test runs in a separate JVM by adding `reuseForks=false` into the surefire plugin's configuration (for the `server/implementation` module)?

```
<plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <configuration>
                    <useFile>false</useFile>
                    <trimStackTrace>false</trimStackTrace>
                    <reuseForks>false</reuseForks>
                </configuration>
            </plugin>
        </plugins>
```

Then you can reintroduce the `if (Boolean.getBoolean("smallrye.graphql.federation.enabled"))` condition in the `Scalars` static initializer

Reply:
Adding `reuseForks=false` worked perfectly, thanks for the idea! Inside `GraphQLScalarTypes`, I also had to change the Federation check to `Boolean.getBoolean("smallrye.graphql.federation.enabled")`, since it seems to be used by the `smallrye-graphql-maven-plugin` when generating the schema.